pkg/process/checks/net.go

```diff
@@ -184,6 +184,56 @@ func getConnectionsByPID(conns *model.Connections) map[int32][]*model.Connection
 	return result
 }
 
+func convertDNSEntry(dnstable map[string]*model.DNSDatabaseEntry, namemap map[string]int32, namedb *[]string, ip string, entry *model.DNSEntry) {
+	dbentry := &model.DNSDatabaseEntry{}
+	for _, name := range entry.Names {
+		if idx, ok := namemap[name]; ok {
+			dbentry.NameIndexes = append(dbentry.NameIndexes, idx)
+		} else {
+			dblen := int32(len(*namedb))
+			*namedb = append(*namedb, name)
+			namemap[name] = dblen
+			dbentry.NameIndexes = append(dbentry.NameIndexes, dblen)
+		}
+
+	}
+	dnstable[ip] = dbentry
+}
+
+func remapDNSStatsByDomain(c *model.Connection, namemap map[string]int32, namedb *[]string, dnslist []string) {
+	old := c.DnsStatsByDomain
```

Comment:
Can we check if the length is `0` and if yes, bail out early without creating the map?